schema_editor/app/scripts/views/login/auth-controller.js

```diff
@@ -0,0 +1,47 @@
+(function() {
+    'use strict';
+
+    /**
+     * @ngInject
+     */
+    function AuthController ($scope, $state, $window, AuthService) {
+
+        $scope.auth = {};
+
+        $scope.alerts = [];
+        $scope.addAlert = function(alertObject) {
+            $scope.alerts.push(alertObject);
+        };
+        $scope.closeAlert = function(index) {
+            $scope.alerts.splice(index, 1);
+        };
+
+        $scope.authenticate = function() {
+            $scope.alerts = [];
+            $scope.authenticated = AuthService.authenticate($scope.auth);
+            $scope.authenticated.then(function(result) {
+                if (result.isAuthenticated) {
+                    // redirect to editor main page, with a full page reload,
+                    // to pick up the newly set credentials
+                    $window.location.href = '/editor/';
```

Comment:
Since both endpoints are valid in development, templating the URL via ansible would still break one or the other. It's just the post-login redirect. Running via grunt changes the endpoint to root; if it just changed the port, this would still redirect correctly. Instead of having to insert some sort of ugly port check here just for `grunt serve`, it would be better if `grunt serve` served on the `/editor/` endpoint, which would be consistent.